File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    def is_prime(num):
        if num<2:
            return False
        for i in range(2,int(num**0.5)+1):
            if num%i==0:
                return False
        return True
    
    def convert_base(num,k):
        temp=''
        while num>0:
            temp+=str(num%k)
            num//=k
        return temp[::-1]
    
    answer=0
    num_list=list(filter(lambda x:x!='' and x!='1',convert_base(n,k).split('0')))
    
    for i in num_list:
        if is_prime(int(i)):
            answer+=1
            
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        // 1. n을 k진법으로 변환한다. 
        String convertN = Integer.toString(n,k);
        // 2. 변환된 수 convert를 0으로 split한다. 
        String[] arr = convertN.split("0+");
        // 3. split된 애들 각각 소수 검사해준다.
        for (String num : arr){
            if (isPrime(Long.parseLong(num))){
                answer++;
            }
        }
        return answer;
    }
    
    private boolean isPrime(long n){
        if (n<2) return false;
        if (n==2) return true;
        if (n%2==0) return false;
        
        for (long i=3;i*i<=n;i=i+2){
            if (n%i==0) return false;
        }
        return true;
    }
}